pkg/apply/task/inv_set_task.go

```diff
@@ -118,3 +122,23 @@ func (i *InvSetTask) Cancel(_ *taskrunner.TaskContext) {}
 
 // StatusUpdate is not supported by the InvSetTask.
 func (i *InvSetTask) StatusUpdate(_ *taskrunner.TaskContext, _ object.ObjMetadata) {}
+
+func (i InvSetTask) getApplyStatus(taskContext *taskrunner.TaskContext) map[object.ObjMetadata]status.ApplyStatus {
+	applyStatus := map[object.ObjMetadata]status.ApplyStatus{}
+	for _, obj := range taskContext.SuccessfulApplies() {
+		applyStatus[obj] = status.ApplySucceeded
+	}
+	for _, obj := range i.PrevInventory.Intersection(taskContext.FailedApplies()) {
+		applyStatus[obj] = status.ApplyFailed
```

Comment:
We should only store references to objects which exist in the cluster. We store objects which failed if they had previously succeeded.

Reply:
The inventory set is not affected. For the ConfigMap inventory, the passed in apply/reconcile status is not used for now. For the inventory object in Kpt, the apply/reconcile status is stored in `.status`.